opentelemetry/src/metrics/mod.rs

```diff
@@ -22,12 +22,12 @@ pub use instruments::{
 pub use meter::{Meter, MeterProvider};
 
 /// A specialized `Result` type for metric operations.
-pub type Result<T> = result::Result<T, MetricsError>;
+pub type MetricResult<T> = result::Result<T, MetricError>;
```

Comment:
I am unsure if this Result/MetricError need to be kept public. For sure, we are removing it from the `opentelemetry` crate (see https://github.com/open-telemetry/opentelemetry-rust/issues/1042).

I believe these Results (public) are only useful when returning a Result from end-user APIs. We have very few such APIs, and most of the time we would be relying on internal logs.
eg of user facing api which benefit from a public Result is ForceFlush(), Shutdown().

It'd be better if we take a step back and see if there would be any actual need of MetricResult in the future, and if not, lets clean it up. And also, keep it in sdk crate only as opentelemetry's API almost never need it.

@pitoniak32 What do you think?

Reply:
I think that makes sense. The only argument I would have in favor of keeping it would be, in my personal project code where I'm doing the setup for these apis I would use these result/error types as the return type from the setup function (if I wasn't already using anyhow or translating to custom errors) just to make things cleaner and avoid doing a Box dyn error. But this is a very specific use case, just wanted to play devils advocate on this one.

Reply:
I would say definitely that the Result types can be removed from the public API but I could see the Errors being useful to keep for users to translate into their own errors, or return from setup functions

Reply:
Would you also do the rename of metrics items as part of this too?

Reply:
No, the rename for `MetricError` and `MetricResult` can go through as part of this PR.